src/rpc/mining.cpp

```diff
@@ -731,7 +731,8 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
 
         if (certSupported) {
             CCoinsViewCache view(pcoinsTip);
-            pblock->hashScTxsCommitment = pblock->BuildScTxsCommitment(view);
+            pblock->hashScTxsCommitment.SetNull();
+            pblock->BuildScTxsCommitment(view, pblock->hashScTxsCommitment);
```

Comment:
If I understand correctly, if we reach this line the commitment tree root should be computed without errors.

What about adding an `assert` on the return value just to be sure that we don't return a valid block with an invalid commitment tree in case of malfunctions?

Reply:
Agree, it could be quite useful. Addressed in e6db47d33

Reply:
Thanks! Sorry for nitpicking here, but given that we may want in the future to completely get rid of all the asserts in the code, what do you think about keeping the call to `BuildScTxsCommitment()` outside of the `assert` and just checking the return code?